### PR TITLE
Add vxid planck layout

### PR DESCRIPTION
**Pull request by vxid**
## Description

Add a custom planck layout

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/planck/keymaps/vxid/readme.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# Planck
```
Comment (mechmerlin) [inline keyboards/planck/keymaps/vxid/readme.md]: This readme is actually intended specifically for your keymap. What you have here is a readme for the keyboard. In here you can put notes on what your keymap does differently than default, how it's built, any quirks, etc etc.

Comment (vxid) [reply keyboards/planck/keymaps/vxid/readme.md]: Hi.
My bad, I added a brief description of my keymap.

File: keyboards/planck/keymaps/vxid/keymap.c [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+
+};
+
+uint32_t layer_state_set_user(uint32_t state) {
+  return update_tri_layer_state(state, _LOWER, _RAISE, _ADJUST);
+}
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+  switch (keycode) {
+    case QWERTY:
```
Comment (drashna) [inline keyboards/planck/keymaps/vxid/keymap.c]: If you only have the QWERTY layer, you can remove this entire section. As well as the keycode, in general

File: keyboards/planck/keymaps/vxid/keymap.c [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+      }
+      return false;
+      break;
+  }
+  return true;
+}
+
+void encoder_update(bool clockwise) {
+    if (clockwise) {
+        register_code(KC_PGDN);
```
Comment (drashna) [inline keyboards/planck/keymaps/vxid/keymap.c]: You can use `tap_code(KC_PGDN);` here instead.  It handles the register/unregister calls, and has a "delay" that you can set, if you have issues with the encoder registering the key properly.

File: keyboards/planck/keymaps/vxid/keymap.c [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+extern keymap_config_t keymap_config;
+
+enum planck_layers {
+  _QWERTY,
+  _LOWER,
+  _RAISE,
+  _ADJUST
+};
+
+enum planck_keycodes {
```
Comment (drashna) [inline keyboards/planck/keymaps/vxid/keymap.c]: If you're not using any of these keycodes, you could remove the entire enum, actually.

**Comment by vxid** [comment]
@drashna @mechmerlin  Thank you for the advice. I updated the keymap and rebased. 
It is ready to merge now if it's okay for you.

Thanks!

**Comment by drashna** [comment]
Thanks!

**Pull request merged**